### PR TITLE
nss-systemd: two minor fixes

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Review by yuwata — APPROVED** [review]
LGTM.

File: src/nss-systemd/nss-systemd.c [hunk 1]
```diff
@@ -312,16 +312,16 @@ enum nss_status _nss_systemd_endgrent(void) {
...
         NSS_ENTRYPOINT_BEGIN;
 
         if (_nss_systemd_is_blocked())
                 return NSS_STATUS_NOTFOUND;
 
-        _cleanup_(pthread_mutex_unlock_assertp) pthread_mutex_t *_l = NULL;
-        int r;
-
-        _l = pthread_mutex_lock_assert(&getpwent_data.mutex);
+        _cleanup_(pthread_mutex_unlock_assertp) pthread_mutex_t *_l =
```
Comment (mrc0mmand) [inline src/nss-systemd/nss-systemd.c:323]: 20:49:35     archlinux_systemd_ci: ../build/src/nss-systemd/nss-systemd.c:323:66: error: unused variable '_l' [-Werror,-Wunused-variable]
20:49:35     archlinux_systemd_ci:         _cleanup_(pthread_mutex_unlock_assertp) pthread_mutex_t *_l =

File: src/nss-systemd/nss-systemd.c [hunk 2]
```diff
@@ -336,19 +336,19 @@ enum nss_status _nss_systemd_setpwent(int stayopen) {
...
         NSS_ENTRYPOINT_BEGIN;
 
         if (_nss_systemd_is_blocked())
                 return NSS_STATUS_NOTFOUND;
 
-        _cleanup_(pthread_mutex_unlock_assertp) pthread_mutex_t *_l = NULL;
-        int r;
-
-        _l = pthread_mutex_lock_assert(&getgrent_data.mutex);
+        _cleanup_(pthread_mutex_unlock_assertp) pthread_mutex_t *_l =
```
Comment (mrc0mmand) [inline src/nss-systemd/nss-systemd.c]:
```
../build/src/nss-systemd/nss-systemd.c:347:66: error: unused variable '_l' [-Werror,-Wunused-variable]
        _cleanup_(pthread_mutex_unlock_assertp) pthread_mutex_t *_l =
                                                                 ^
```

File: src/nss-systemd/nss-systemd.c (hunk 1 above)
Comment (mrc0mmand) [inline src/nss-systemd/nss-systemd.c]: ditto

File: src/nss-systemd/nss-systemd.c [hunk 3]
```diff
@@ -372,9 +372,8 @@ enum nss_status _nss_systemd_getpwent_r(
         if (_nss_systemd_is_blocked())
                 return NSS_STATUS_NOTFOUND;
 
-        _cleanup_(pthread_mutex_unlock_assertp) pthread_mutex_t *_l = NULL;
-
-        _l = pthread_mutex_lock_assert(&getpwent_data.mutex);
+        _cleanup_(pthread_mutex_unlock_assertp) pthread_mutex_t *_l
```
Comment (mrc0mmand) [inline src/nss-systemd/nss-systemd.c]: ditto

File: src/nss-systemd/nss-systemd.c [hunk 4]
```diff
@@ -419,9 +418,8 @@ enum nss_status _nss_systemd_getgrent_r(
         if (_nss_systemd_is_blocked())
                 return NSS_STATUS_NOTFOUND;
 
-        _cleanup_(pthread_mutex_unlock_assertp) pthread_mutex_t *_l = NULL;
-
-        _l = pthread_mutex_lock_assert(&getgrent_data.mutex);
+        _cleanup_(pthread_mutex_unlock_assertp) pthread_mutex_t *_l =
```
Comment (mrc0mmand) [inline src/nss-systemd/nss-systemd.c]: ditto

**Review by mrc0mmand — CHANGES_REQUESTED** [review]

File: src/nss-systemd/nss-systemd.c (hunk 1 above)
Comment (keszybz) [reply src/nss-systemd/nss-systemd.c]: Yeah, using a separate variable here is a trick to trick clang into thinking that the variable is used.

**Review by keszybz — COMMENTED** [review]
I'll force-push with the middle commit dropped, to restart the CI.

**Comment by poettering** [comment]
Hmm, I tried to trick llvm into accepting this, by adding `_used_` to it, but then gcc starts complaining:

```
../src/nss-systemd/nss-systemd.c: In function ‘_nss_systemd_setpwent’:
../src/nss-systemd/nss-systemd.c:323:9: warning: ‘used’ attribute ignored [-Wattributes]
  323 |         _cleanup_(pthread_mutex_unlock_assertp) pthread_mutex_t *_l _used_ =
```

:-(

**Pull request merged**